### PR TITLE
fix: Better weight decay exclusion in `run_*_no‑trainer.py` examples

### LINKED ISSUE #42754: Excluding weight decay not working properly on most LMs
The problem is consistent across all three files:
https://github.com/huggingface/transformers/blob/471d7ce9abbb3bc1b3bab673367378f9dbc3caac/examples/pytorch/language-modeling/run_clm_no_trainer.py#L518-L528
https://github.com/huggingface/transformers/blob/471d7ce9abbb3bc1b3bab673367378f9dbc3caac/examples/pytorch/language-modeling/run_fim_no_trainer.py#L710-L720
https://github.com/huggingface/transformers/blob/471d7ce9abbb3bc1b3bab673367378f9dbc3caac/examples/pytorch/language-modeling/run_mlm_no_trainer.py#L544-L554
Many LMs use patterns like `[r"layernorm", r"rmsnorm", r"(?:^|\.|_)norm(?:$|\.|_)", r"(?:^|\.|_)ln(?:$|\.|_)"]`, so the simple `no_decay` patterns here won't exclude many normalization layers. I also suggest excluding embeddings and `r"(?:^|\.)wpe(?:$|\.)"` to exclude input/output/positional embeddings.
Could consider using/modifying from the code here:
https://github.com/huggingface/transformers/blob/471d7ce9abbb3bc1b3bab673367378f9dbc3caac/src/transformers/trainer.py#L1213-L1227

### PR DESCRIPTION
**Pull request by casinca**
# What does this PR do?
Hello,

This is a complementary PR to #42769 (not yet pushed, at the time of writing) in order to fix #42754

Tests passed (at least the ones with a test.)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Was discussed with @Rocketknight1 in #42769

### CODE REVIEW

**Review by Rocketknight1 — APPROVED** [review]
Yep, LGTM!
